src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,183 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.Constant;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidUserAgeException;
+import core.basesyntax.exception.InvalidUserLoginException;
+import core.basesyntax.exception.InvalidUserPasswordException;
+import core.basesyntax.exception.UserAlreadyExistsException;
+import core.basesyntax.exception.UserValidationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+    private User validUser = new User();
+
+    @BeforeEach
+    public void each() {
+        Storage.people.clear();
+        validUser.setLogin(Constant.VALID_LOGIN);
+        validUser.setAge(Constant.VALID_AGE);
+        validUser.setPassword(Constant.VALID_PASSWORD);
+    }
+
+    @Test
+    void register_validUser_Ok() {
+        assertEquals(validUser, registrationService.register(validUser));
+
+        User validUser2 = new User();
+        validUser2.setLogin(Constant.VALID_LOGIN2);
+        validUser2.setAge(Constant.VALID_AGE2);
+        validUser2.setPassword(Constant.VALID_PASSWORD2);
+        assertEquals(validUser2, registrationService.register(validUser2));
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(UserValidationException.class,
+                () -> registrationService.register(null)
+        );
+    }
+
+    @Test
+    void register_existUser_notOk() {
+        registrationService.register(validUser);
```

Comment:
I think @yevheniisynytsia made a mistake here, we should be using the direct Storage or dao here as it is written in the common mistakes, and I think it was also discussed on QnA, please check the recording of the one on 14:00 for more details,